switchbot/__init__.py

```diff
@@ -51,6 +62,19 @@ def _disconnect(self) -> None:
         finally:
             self._device = None
 
+    def _commandkey(self, key) -> str:
+        key = ""
+        key_suffix = PRESS_KEY_SUFFIX
+        if key == ON_KEY:
+            key_suffix = ON_KEY_SUFFIX
+        elif key == OFF_KEY:
+            key_suffix = OFF_KEY_SUFFIX
+        if self._password_encoded is not None:
+            key = KEY_PASSWORD_PREFIX + self._password_encoded + key_suffix
+        else:
+            key = KEY_PREFIX + key_suffix
+        return key
```

Comment:
Why is the key modified if no password is set?

Reply:
I'll get back to you when I've tested more

Reply:
Good catch.

I've removed it and committed. Have tested, but note that the switchbot need to be set in switch mode or press mode from the app in order to use the press, turn_on and turn_off methods.